src/bernstein.jl

```diff
@@ -0,0 +1,79 @@
+export BernsteinBasis
+
+struct BernsteinBasis{N} <: AbstractP4MLBasis where {N} end
+
+BernsteinBasis(N::Integer) = BernsteinBasis{N}()
+
+Base.length(basis::BernsteinBasis{N}) where {N} = N
+
+natural_indices(basis::BernsteinBasis) = 0:length(basis)-1
+
+_valtype(basis::BernsteinBasis, T::Type{<:Real}) = T
+
+_generate_input(basis::BernsteinBasis) = rand()  
+
+
+using StaticArrays
+
+function binom_svec(n::Int)
```

Comment:
This should be type-unstable. And is `SVector` needed here? Could you just replace it with 
```julia
ntuple(k -> binomial(N-1, k-1), N)
```

Reply:
I also wonder whether there is a fast way to generate all binomial coefficients at once, precompiled, something like this. 
```julia
static_binomial(::Val{1}) = (1,)
static_binomial(::Val{2}) = (1, 1) 
@generated function static_binomial(::Val{N}) where {N}
          b = [ binomial(N-1, k) for k = 0:N-1 ]
          code = Meta.parse( "( " * prod("$(b[i]), " for i = 1:N) * " )"  )
          quote
            $code
          end
       end   
```